tests/server/wildfly/src/test/java/org/hawkular/apm/tests/wildfly/zipkin/SpanAnalyticsITest.java

```diff
@@ -0,0 +1,253 @@
+import org.hawkular.apm.tests.common.Wait;
+import org.hawkular.apm.tests.wildfly.AbstractITest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public class SpanAnalyticsITest extends AbstractITest {
+
+    private static AnalyticsServiceRESTClient analyticsService;
+
+    @BeforeClass
+    public static void beforeClass() {
+        analyticsService = new AnalyticsServiceRESTClient();
+        analyticsService.setUsername(HAWKULAR_APM_USERNAME);
+        analyticsService.setPassword(HAWKULAR_APM_PASSWORD);
+    }
+
+    @Before
+    public void beforeTest() {
+        analyticsService.clear(null);
+    }
+
+    @Test
+    public void testCompletionTime() throws IOException {
+        BinaryAnnotation urlRootBinAnn = new BinaryAnnotation();
+        urlRootBinAnn.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        urlRootBinAnn.setValue("http://localhost/root");
+        urlRootBinAnn.setType(AnnotationType.STRING);
+
+        Span rootSpan = new Span(Arrays.asList(urlRootBinAnn),
+                SpanServiceITest.serverAnnotations(
+                        SpanServiceITest.millisToMicroSeconds(1), SpanServiceITest.millisToMicroSeconds(30)));
+        rootSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(1L));
+        rootSpan.setDuration(SpanServiceITest.calculateDuration(1L, 30L));
+        rootSpan.setTraceId("root");
+        rootSpan.setId("root");
+        rootSpan.setName("get");
+
+        BinaryAnnotation urlServiceABinAnn = new BinaryAnnotation();
+        urlServiceABinAnn.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        urlServiceABinAnn.setValue("http://localhost/serviceA");
+
+        Span clientSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.clientAnnotation(SpanServiceITest.millisToMicroSeconds(5),
+                        SpanServiceITest.millisToMicroSeconds(15)));
+        clientSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(5));
+        clientSpan.setDuration(SpanServiceITest.calculateDuration(5L, 15L));
+        clientSpan.setTraceId(rootSpan.getTraceId());
+        clientSpan.setId("2");
+        clientSpan.setParentId("root");
+        clientSpan.setName("post");
+
+        Span serverSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.serverAnnotations(
+                        SpanServiceITest.millisToMicroSeconds(7), SpanServiceITest.millisToMicroSeconds(10)));
+        serverSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(7));
+        serverSpan.setDuration(SpanServiceITest.calculateDuration(7L, 10L));
+        serverSpan.setTraceId(rootSpan.getTraceId());
+        serverSpan.setId("2");
+        serverSpan.setParentId("root");
+        serverSpan.setName("post");
+
+        post(Server.Zipkin, "/spans", null, Arrays.asList(rootSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(clientSpan));
```

Comment:
May be good to post root and client span together but server span separately, as this better reflects how spans could be reported.